crates/iota-indexer/src/indexer_reader.rs

```diff
@@ -1582,40 +1585,33 @@ impl IndexerReader {
     }
 
     pub fn get_latest_move_call_metrics(&self) -> IndexerResult<MoveCallMetrics> {
-        let latest_3d_move_call_metrics = run_query!(&self.pool, |conn| {
-            move_call_metrics::table
-                .filter(move_call_metrics::dsl::day.eq(3))
-                .order(move_call_metrics::dsl::id.desc())
-                .limit(10)
-                .load::<QueriedMoveCallMetrics>(conn)
-        })?;
-        let latest_7d_move_call_metrics = run_query!(&self.pool, |conn| {
-            move_call_metrics::table
-                .filter(move_call_metrics::dsl::day.eq(7))
-                .order(move_call_metrics::dsl::id.desc())
-                .limit(10)
-                .load::<QueriedMoveCallMetrics>(conn)
-        })?;
-        let latest_30d_move_call_metrics = run_query!(&self.pool, |conn| {
-            move_call_metrics::table
-                .filter(move_call_metrics::dsl::day.eq(30))
-                .order(move_call_metrics::dsl::id.desc())
-                .limit(10)
-                .load::<QueriedMoveCallMetrics>(conn)
-        })?;
+        let fetch_metrics = |conn: &mut PgConnection,
+                             day_value: i64|
+         -> diesel::QueryResult<Vec<(MoveFunctionName, usize)>> {
+            let query = "
+            SELECT id, epoch, day, move_package, move_module, move_function, count
+            FROM move_call_metrics
+            WHERE day = $1
+              AND epoch = (SELECT MAX(epoch) FROM move_call_metrics WHERE day = $1)
+            ORDER BY count DESC
+            LIMIT 10
+        ";
+
+            sql_query(query)
+                .bind::<BigInt, _>(day_value)
+                .load::<QueriedMoveCallMetrics>(conn)?
+                .into_iter()
+                .map(|m| {
+                    m.try_into()
+                        .map_err(|e| diesel::result::Error::DeserializationError(Box::new(e)))
+                })
+                .collect()
+        };
```

Comment:
Nit: Move to a dedicated function. This will improve readability, testing, reuseability.

Reply:
Fixed with https://github.com/iotaledger/iota/pull/5648/commits/961202c2fb4be9b8f2547e6488624601536d4cb4